retail-forecast/materializers/prophet_materializer.py

```diff
@@ -0,0 +1,156 @@
+import json
+import os
+from typing import Any, Dict, Type
+
+from prophet import Prophet
+from prophet.serialize import model_from_json, model_to_json
+from zenml.enums import ArtifactType
+from zenml.materializers.base_materializer import BaseMaterializer
+
+
+class ProphetMaterializer(BaseMaterializer):
+    """Materializer for Prophet models."""
+
+    ASSOCIATED_TYPES = (Prophet, dict)
+    ASSOCIATED_ARTIFACT_TYPE = ArtifactType.MODEL
+
+    def load(self, data_type: Type[Any]) -> Any:
+        """Load a Prophet model or dictionary of Prophet models from storage.
+
+        Args:
+            data_type: The data type to load.
+
+        Returns:
+            A Prophet model or dictionary of Prophet models.
+        """
+        # Check if we're loading a dictionary
+        if data_type == dict:
+            # Path to the keys file
+            keys_path = os.path.join(self.uri, "keys.json")
+
+            # Load the keys
+            with self.artifact_store.open(keys_path, "r") as f:
+                keys = json.load(f)
+
+            # Load each model
+            result = {}
+            for key in keys:
+                model_dir = os.path.join(self.uri, f"models/{key}")
```

Comment:
Won't the `/` not work with Windows? Probably need to specify models as something separate?